netpyne_ui/experiments.py

```diff
@@ -40,13 +64,122 @@ def edit_experiment(name: str, experiment: dict):
     _add_experiment(updated_exp)
 
 
+def replace_current_with(name: str):
+    exp = _get_by_name(name)
+    if not exp:
+        raise ExperimentsError(f"Experiment with name {name} does not exist")
+
+    next_name = utils.get_next_file_name(constants.EXPERIMENTS_FOLDER_PATH, name)
+    new_exp = model.Experiment(
+        name=next_name,
+        state=model.ExperimentState.DESIGN,
+        params=exp.params,
+        seed=exp.seed,
+        initConfig=exp.initConfig,
+        method=exp.method
+    )
+
+    current = get_current()
+    if current:
+        remove_experiment(current.name)
+
+    _add_experiment(new_exp)
+    return
+
+
+def get_current() -> model.Experiment:
+    return next(
+        (exp for exp in model.experiments if exp.state == model.ExperimentState.DESIGN),
+        None
+    )
+
+
 def _add_experiment(experiment: model.Experiment):
     if _get_by_name(experiment.name):
         raise ExperimentsError(f"Experiment {experiment.name} already exists")
 
+    _generate_trials(experiment)
     model.experiments.append(experiment)
 
 
 def _get_by_name(name: str) -> model.Experiment:
     experiment = next((e for e in model.experiments if e.name == name), None)
     return experiment
+
+
+def _scan() -> [model.Experiment]:
```

Comment:
A less generic name would be better for such a key function. Something like _scan_experiments_in_directory?
I'd also add make the folder a parameter such that it becomes unit-testable

Reply:
yeah agree, updated it!